framework/Models/PostProcessors/Validations/PhysicsGuidedCoverageMapping.py

```diff
@@ -0,0 +1,218 @@
+      results to specified dataObject
+      @ In, inputIn, list, dictionary of data to process
+      @ Out, outputDict, dict, dictionary containing the post-processed results
+    """
+    dataDict = {self.getDataSetName(data): data for _, _, data in inputIn['Data']}
+    pivotParameter = self.pivotParameter
+    names = [self.getDataSetName(inp[-1]) for inp in inputIn['Data']]
+    if len(inputIn['Data'][0][-1].indexes) and self.pivotParameter is None:
+      if 'dynamic' not in self.dynamicType: #self.model.dataType:
+        self.raiseAnError(IOError, "The validation algorithm '{}' is not a dynamic model but time-dependent data has been inputted in object {}".format(self._type, inputIn['Data'][0][-1].name))
+    evaluation ={k: np.atleast_1d(val) for k, val in  self._evaluate(dataDict, **{'dataobjectNames': names}).items()}
+
+    if pivotParameter:
+      if len(inputIn['Data'][0][-1]['time']) != len(list(evaluation.values())[0]):
+        self.raiseAnError(RuntimeError, "The pivotParameter value '{}' has size '{}' and validation output has size '{}'".format( len(dataSets[0][self.pivotParameter]), len(evaluation.values()[0])))
+      if pivotParameter not in evaluation:
+        evaluation[pivotParameter] = inputIn['Data'][0][-1]['time']
+    return evaluation
+
+  ### utility functions
+  def _evaluate(self, datasets, **kwargs):
+    """
+      Main method to "do what you do".
+      @ In, datasets, list, list of datasets (data1,data2,etc.) to used.
+      @ In, kwargs, dict, keyword arguments
+      @ Out, outputDict, dict, dictionary containing the results {"feat"_"target"_"metric_name":value}
+    """
+    names = kwargs.get('dataobjectNames')
+    outputDict = {}
+    for feat, targ in zip(self.features, self.targets):
+      featData = self._getDataFromDataDict(datasets, feat, names)
+      targData = self._getDataFromDataDict(datasets, targ, names)
+
+      # Standardization
+      yExp = np.array(featData[0]).reshape(len(featData[0]))
+      yApp = np.array(targData[0]).reshape(len(targData[0]))
+      yExpRef = np.mean(yExp)
+      yAppRef = np.mean(yApp)
+      yExpStd = (yExp-yExpRef)/yExpRef
+      yAppStd = (yApp-yAppRef)/yAppRef
+
+      # Kernel Desnity Estimation
+      m1 = yExpStd[:]
+      m2 = yAppStd[:]
+      xmin = m1.min()
+      xmax = m1.max()
+      ymin = m2.min()
+      ymax = m2.max()
+
+      binKDE = 200j
+      X, Y = np.mgrid[xmin:xmax:binKDE, ymin:ymax:binKDE]
+      psts = np.vstack([X.ravel(), Y.ravel()])
+      vals = np.vstack([m1, m2])
+      # kernel
+      knl = stats.gaussian_kde(vals)
+      Z = np.reshape(knl(psts).T, X.shape)
+
+      # Virtual Measurement
+      msrMean = 0.0
+      msrStd = 0.01*np.std(yAppStd)
+      msrNumSmpl = 1000
+      randomUtils.randomSeed(0)
+      yMsr = msrStd * randomUtils.randomNormal(size=msrNumSmpl) + msrMean
+      binMsr = msrNumSmpl//20
+      yMsrPdf, yMsrBin = np.histogram(yMsr, binMsr, range=(xmin, xmax), density=True)
```

Comment:
Data for measurement should be provided through csv file, which can be loaded by raven, instead of hard-coded here.

Reply:
I don't know how to add another input variable here, other methods in this validation gate only requires Features and Targets, no measurement information required.